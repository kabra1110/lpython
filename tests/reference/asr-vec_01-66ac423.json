{
    "basename": "asr-vec_01-66ac423",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/vec_01.py",
    "infile_hash": "b22cabe2248cf58c7ffaad83dd8dc5b2433f244ee23b0703b536547b",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-vec_01-66ac423.stdout",
    "stdout_hash": "49151993950e6ac89ab0d0a39cc2d0ff258186636fb759f9cdd6dc9d",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}