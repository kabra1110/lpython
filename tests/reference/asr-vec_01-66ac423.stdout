(TranslationUnit
    (SymbolTable
        1
        {
            _global_symbols:
                (Module
                    (SymbolTable
                        205
                        {
                            _lpython_main_program:
                                (Function
                                    (SymbolTable
                                        204
                                        {
                                            
                                        })
                                    _lpython_main_program
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [loop_vec]
                                    []
                                    [(SubroutineCall
                                        205 loop_vec
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            loop_vec:
                                (Function
                                    (SymbolTable
                                        199
                                        {
                                            a:
                                                (Variable
                                                    199
                                                    a
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 8)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 9216 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            b:
                                                (Variable
                                                    199
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 8)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 9216 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    199
                                                    i
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    loop_vec
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(DoLoop
                                        ()
                                        ((Var 199 i)
                                        (IntegerConstant 0 (Integer 4))
                                        (IntegerBinOp
                                            (IntegerConstant 9216 (Integer 4))
                                            Sub
                                            (IntegerConstant 1 (Integer 4))
                                            (Integer 4)
                                            (IntegerConstant 9215 (Integer 4))
                                        )
                                        (IntegerConstant 1 (Integer 4)))
                                        [(=
                                            (ArrayItem
                                                (Var 199 b)
                                                [(()
                                                (Var 199 i)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            (RealConstant
                                                5.000000
                                                (Real 8)
                                            )
                                            ()
                                        )]
                                    )
                                    (DoLoop
                                        ()
                                        ((Var 199 i)
                                        (IntegerConstant 0 (Integer 4))
                                        (IntegerBinOp
                                            (IntegerConstant 9216 (Integer 4))
                                            Sub
                                            (IntegerConstant 1 (Integer 4))
                                            (Integer 4)
                                            (IntegerConstant 9215 (Integer 4))
                                        )
                                        (IntegerConstant 1 (Integer 4)))
                                        [(=
                                            (ArrayItem
                                                (Var 199 a)
                                                [(()
                                                (Var 199 i)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            (ArrayItem
                                                (Var 199 b)
                                                [(()
                                                (Var 199 i)
                                                ())]
                                                (Real 8)
                                                RowMajor
                                                ()
                                            )
                                            ()
                                        )]
                                    )
                                    (DoLoop
                                        ()
                                        ((Var 199 i)
                                        (IntegerConstant 0 (Integer 4))
                                        (IntegerBinOp
                                            (IntegerConstant 9216 (Integer 4))
                                            Sub
                                            (IntegerConstant 1 (Integer 4))
                                            (Integer 4)
                                            (IntegerConstant 9215 (Integer 4))
                                        )
                                        (IntegerConstant 1 (Integer 4)))
                                        [(Assert
                                            (RealCompare
                                                (ArrayItem
                                                    (Var 199 a)
                                                    [(()
                                                    (Var 199 i)
                                                    ())]
                                                    (Real 8)
                                                    RowMajor
                                                    ()
                                                )
                                                Eq
                                                (RealConstant
                                                    5.000000
                                                    (Real 8)
                                                )
                                                (Logical 4)
                                                ()
                                            )
                                            ()
                                        )]
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    _global_symbols
                    []
                    .false.
                    .false.
                ),
            lpython_builtin:
                (IntrinsicModule lpython_builtin),
            main_program:
                (Program
                    (SymbolTable
                        203
                        {
                            _lpython_main_program:
                                (ExternalSymbol
                                    203
                                    _lpython_main_program
                                    205 _lpython_main_program
                                    _global_symbols
                                    []
                                    _lpython_main_program
                                    Public
                                )
                        })
                    main_program
                    [_global_symbols]
                    [(SubroutineCall
                        203 _lpython_main_program
                        ()
                        []
                        ()
                    )]
                ),
            numpy:
                (Module numpy)
        })
    []
)
